internal/wasmldr/src/loader/configured/mod.rs

```diff
@@ -53,21 +51,40 @@ impl Loader<Configured> {
     }
 
     pub fn next(self) -> Result<Loader<Requested>> {
-        // TODO:
-        //   1. get attestation
-        //   2. get intermediate cert
-        //   3. choose key type based on technology
-        //   4. generate tech-specific attestation extension request
+        // TODO: get intermediate cert
+
+        let platform = Platform::get()?;
+        let cert_algo = match platform.technology() {
+            Technology::Snp => SECP_384_R_1,
+            Technology::Sgx => SECP_256_R_1,
+            Technology::Kvm => SECP_256_R_1,
+            _ => return Err(anyhow!("unknown platform")),
+        };
 
         // Generate a keypair.
-        let raw = PrivateKeyInfo::generate(SECP_256_R_1)?;
+        let raw = PrivateKeyInfo::generate(cert_algo)?;
         let pki = PrivateKeyInfo::from_der(raw.as_ref())?;
 
+        let mut key_hash = [0u8; 64];
+
+         match platform.technology() {
+            Technology::Snp => {
+                let hash = Sha384::digest(raw.as_slice());
```

Comment:
@rjzak @npmccallum out of curiousity, what does hashing of the private key give?

Reply:
@rjzak This needs to be a hash of the PUBLIC key. Not the private key.

Reply:
@rjzak Convert the `PrivateKeyInfo` into a `SubjectPublicKeyInfo` and then encode the SPKI to DER and hash that.